index.js

```diff
@@ -44,9 +55,15 @@ class ExecutorQueue extends Executor {
      * @return {Promise}
      */
     _start(config) {
-        return this.breaker.runCommand('connect')
-            // Note: arguments to enqueue are [queue name, job type, array of args]
-            .then(() => this.breaker.runCommand('enqueue', this.buildQueue, 'start', [config]));
+        return this.queueBreaker.runCommand('connect')
+            // Store the config in redis
+            .then(() => this.redisBreaker.runCommand('hset', this.buildConfigTable,
+                config.buildId, JSON.stringify(config)))
+            // Note: arguments to enqueue are [queue name, job name, array of args]
+            .then(() => this.queueBreaker.runCommand('enqueue', this.buildQueue, 'start', [{
+                buildId: config.buildId,
+                annotations: config.annotations || ''
```

Comment:
Right. Or if there's some operation that looks up some metadata regarding a build (e.g., job information), then we can include that in the parameter.